remote/remote_state_s3.go

```diff
@@ -381,20 +390,207 @@ func createS3BucketIfNecessary(s3Client *s3.S3, config *ExtendedRemoteStateConfi
+			return err
+		}
+	}
+
+	if configBucket.SSEEncryption != "" {
+		if config.SkipBucketSSEncryption {
+			terragruntOptions.Logger.Debugf("Server-Side Encryption is disabled for the remote state AWS S3 bucket %s using 'skip_bucket_ssencryption' config.", config.remoteStateConfigS3.Bucket)
+		} else if err := EnableSSEForS3BucketWide(s3Client, config, terragruntOptions); err != nil {
+			return err
+		}
+	}
+
+	if configBucket.RootAccess != "" {
+		if config.SkipBucketRootAccess {
+			terragruntOptions.Logger.Debugf("Root access is disabled for the remote state S3 bucket %s using 'skip_bucket_root_access' config.", config.remoteStateConfigS3.Bucket)
+		} else if err := EnableRootAccesstoS3Bucket(s3Client, config, terragruntOptions); err != nil {
+			return err
+		}
+	}
+
+	if configBucket.EnforcedTLS != "" {
+		if config.SkipBucketEnforcedTLS {
+			terragruntOptions.Logger.Debugf("Enforced TLS is disabled for the remote state AWS S3 bucket %s using 'skip_bucket_enforced_tls' config.", config.remoteStateConfigS3.Bucket)
+		} else if err := EnableEnforcedTLSAccesstoS3Bucket(s3Client, config, terragruntOptions); err != nil {
+			return err
+		}
+	}
+
+	if configBucket.AccessLogging != "" {
+		if config.SkipBucketAccessLogging {
+			terragruntOptions.Logger.Debugf("Access logging is disabled for the remote state AWS S3 bucket %s using 'skip_bucket_access_logging' config.", config.remoteStateConfigS3.Bucket)
+		} else {
+			if config.AccessLoggingBucketName != "" {
+				terragruntOptions.Logger.Debugf("Enabling bucket-wide Access Logging on AWS S3 bucket %s - using as TargetBucket %s", config.remoteStateConfigS3.Bucket, config.AccessLoggingBucketName)
+
+				if err := CreateLogsS3BucketIfNecessary(s3Client, aws.String(config.AccessLoggingBucketName), terragruntOptions); err != nil {
+					terragruntOptions.Logger.Errorf("Could not create logs bucket %s for AWS S3 bucket %s", config.AccessLoggingBucketName, config.remoteStateConfigS3.Bucket)
+					return err
+				}
+
+				if err := EnablePublicAccessBlockingForS3Bucket(s3Client, config.AccessLoggingBucketName, terragruntOptions); err != nil {
+					return err
+				}
+
+				if err := EnableAccessLoggingForS3BucketWide(s3Client, &config.remoteStateConfigS3, terragruntOptions, config.AccessLoggingBucketName, config.AccessLoggingTargetPrefix); err != nil {
+					return err
+				}
+			} else {
+				terragruntOptions.Logger.Debugf("Access Logging is disabled for the remote state AWS S3 bucket %s", config.remoteStateConfigS3.Bucket)
+			}
+		}
+	}
+
+	if configBucket.PublicAccess != "" {
+		if err := EnablePublicAccessBlockingForS3Bucket(s3Client, config.remoteStateConfigS3.Bucket, terragruntOptions); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+type S3BucketConfig struct {
+	Versioning    string
+	SSEEncryption string
+	RootAccess    string
+	EnforcedTLS   string
+	AccessLogging string
+	PublicAccess  string
+}
```

Comment:
It looks like value of the string is immaterial in `configBucket`, and is only used to track if these attributes need updating. Is that true?

If so, these should be `bool` instead of `string` and the struct should be renamed to something like `S3BucketUpdatesRequired`.

Reply:
@yorinasub17 Hi, it does make sense. I made the suggested adjustment and the necessary corrections. Let me know if I'm missing something